File: backend/src/main/java/at/ac/tuwien/sepr/groupphase/backend/entity/Subject.java
package at.ac.tuwien.sepr.groupphase.backend.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Subject {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(nullable = false, length = 255)
    private String title;
    @Column(nullable = false, length = 255)
    private String type;
    @Column(nullable = false, length = 255)
    private String number;
    @Column(nullable = false, length = 255)
    private String semester;
    @Column(nullable = false, length = 255)
    private String url;
    @Column(length = 1000)
    private String description;
}

File: frontend/src/app/dtos/subject.ts
export class SubjectCreateDto {
  title: String;
  type: String;
  number: String;
  semester: String;
  url: String;
  description: String;
}

export class SubjectDetailDto {
  title: String;
  type: String;
  number: String;
  semester: String;
  url: String;
  description: String;
  id: number;
}
